apps/passport-client/src/dispatch.ts

```diff
@@ -1035,50 +1036,41 @@ async function doSync(
         state.pcds,
         state.credentialCache
       );
-
-      /**
-       * Because the Email PCD is used as a credential for other feeds, it is
-       * necessary to ensure that the Email PCD is present before polling other
-       * feeds.
-       * We already fetch the Email PCD in {@link finishAccountCreation()}, so
-       * it *should* be present in the PCD collection already. However, there
-       * may have been an intermittent failure (e.g. due to connectivity issues
-       * or the restart of the Zupass server during a deployment). In this
-       * case, we try again here, before continuing to fetch the other feeds.
-       * If there is already an Email PCD then we can skip this step.
-       */
-      if (state.pcds.getPCDsByType(EmailPCDTypeName).length === 0) {
-        console.log(
-          "[SYNC] email PCD not found, attempting to poll Email PCD subscription"
-        );
-        const emailPCDSubscription = state.subscriptions.findSubscription(
-          ZUPASS_FEED_URL,
-          ZupassFeedIds.Email
-        );
-        if (emailPCDSubscription) {
-          console.log("[SYNC] Email PCD subscription found, polling");
-          const emailPCDActions =
-            await state.subscriptions.pollSingleSubscription(
-              emailPCDSubscription,
-              credentialManager
-            );
-          console.log(
-            `[SYNC] Fetched ${emailPCDActions.length} actions from Email PCD feed`
-          );
-          await applyActions(state.pcds, emailPCDActions);
-        }
-      }
       console.log("[SYNC] initalized credentialManager", credentialManager);
+
       const stats = { nActions: 0, nFeeds: 0 };
+      const onSubscriptionResult = async (
+        actions: SubscriptionActions
+      ): Promise<void> => {
+        await applyActions(state.pcds, [actions]);
+        await savePCDs(state.pcds);
+        await saveSubscriptions(state.subscriptions);
+        stats.nActions += actions.actions.length;
+        stats.nFeeds++;
+      };
+
+      // first, always poll the email feed
+      await state.subscriptions.pollEmailSubscription(
+        ZUPASS_FEED_URL,
+        credentialManager,
+        onSubscriptionResult
+      );
```

Comment:
important! behavior is correct, could be optimized

Reply:
What could be optimized?

Reply:
I think i was thinking about ordering, but i forget at this point. the thing that's important is that this happens first, moreso than whether it's optimal or not.